lib/delegator.rb

```diff
@@ -0,0 +1,18 @@
+class Delegator
+    def initialize(parsers)
+        # @parsers = args.fetch(:parsers)
+        @parsers = parsers
+    end
+
+    def delegate(input, game)
+        if contains_parser(input, game)
```

Comment:
https://medium.com/the-code-review/en-garde-how-you-can-refactor-nested-ifs-with-guard-clauses-in-ruby-1ff728b7a0cd would streamline this